File: back-end/src/main/java/com/huyphan/mappers/SearchCriteriaMapper.java
package com.huyphan.mappers;

import com.huyphan.dtos.SearchCriteriaDto;
import com.huyphan.models.SearchCriteria;
import com.huyphan.utils.SearchCriteriaNameGetter;

/** Maps from search criteria DTO to domain model. */
public interface SearchCriteriaMapper
		extends FromDtoMapper<SearchCriteriaDto, SearchCriteria>, SearchCriteriaNameGetter {
}

File: front-end/src/app/core/services/mappers/search-criteria-mapper.ts
import { SearchCriteriaName } from '../../enums/search-criteria-name';
import { SearchCriteria } from '../../models/search-criteria';
import { SearchCriteriaDto } from '../dtos/search-criteria-dto';

import { IMapperToDto } from './mapper';

/** Mapper that maps search criteria value from domain model to DTO. */
export interface SearchCriteriaMapper<T>
  extends IMapperToDto<SearchCriteriaDto<T>, SearchCriteria> {
  /** Returns the search criteria name that this mapper corresponds to.
   */
  getSearchCriteriaName(): SearchCriteriaName;
}
